src/service/resources-impl.js

```diff
@@ -560,12 +566,18 @@ export class Resources {
     // During prerender mode, don't build elements that aren't allowed to be
     // prerendered. This avoids wasting our prerender build quota.
     // See grantBuildPermission() for more details.
-    const shouldBuildResource =
-        this.viewer_.getVisibilityState() != VisibilityState.PRERENDER
-        || resource.prerenderAllowed();
+    const isPrerendering =
+        this.viewer_.getVisibilityState() === VisibilityState.PRERENDER;
+    const shouldBuildResource = !isPrerendering || resource.prerenderAllowed();
 
     if (buildingEnabled && shouldBuildResource) {
       if (this.documentReady_) {
+        if (isPrerendering &&
+            this.viewer_.getPrerenderMode() === PrerenderMode.DFS) {
+          this.dfsPrerender_();
+          return;
+        }
```

Comment:
Don't we also need to prevent the normal path of resource building if DFS mode is enabled?